pandas/core/arrays/masked.py

```diff
@@ -159,24 +168,29 @@ def _hasna(self) -> bool:
         # source code using it..
         return self._mask.any()
 
-    def isna(self):
+    def isna(self) -> np.ndarray:
         return self._mask
 
     @property
     def _na_value(self):
         return self.dtype.na_value
 
     @property
-    def nbytes(self):
+    def nbytes(self) -> int:
         return self._data.nbytes + self._mask.nbytes
 
     @classmethod
-    def _concat_same_type(cls, to_concat):
+    def _concat_same_type(cls: Type[BaseMaskedArrayT], to_concat) -> BaseMaskedArrayT:
         data = np.concatenate([x._data for x in to_concat])
         mask = np.concatenate([x._mask for x in to_concat])
         return cls(data, mask)
 
-    def take(self, indexer, allow_fill=False, fill_value=None):
+    def take(
+        self: BaseMaskedArrayT,
```

Comment:
We otherwise don't type self, or do we? 
(self is always the type of the class, no?)